py/torch_tensorrt/dynamo/compile.py

```diff
@@ -0,0 +1,240 @@
+    use_python_runtime=USE_PYTHON_RUNTIME,
+    **kwargs,
+):
+    if debug:
+        logger.setLevel(logging.DEBUG)
+
+    logger.warn(
+        "The Dynamo backend is an experimental feature, for which only the "
+        + "following arguments are supported: "
+        + "{enabled_precisions, debug, workspace_size, min_block_size, "
+        + "torch_executed_ops, pass_through_build_failures}"
+    )
+
+    if not isinstance(inputs, collections.abc.Sequence):
+        inputs = [inputs]
+
+    inputs = prepare_inputs(inputs, prepare_device(device))
+
+    if (
+        torch.float16 in enabled_precisions
+        or torch_tensorrt.dtype.half in enabled_precisions
+    ):
+        lower_precision = LowerPrecision.FP16
+    elif (
+        torch.float32 in enabled_precisions
+        or torch_tensorrt.dtype.float in enabled_precisions
+    ):
+        lower_precision = LowerPrecision.FP32
+    elif len(enabled_precisions) == 0:
+        logger.info(f"No precision specified, defaulting to {PRECISION}")
+        lower_precision = PRECISION
+    else:
+        raise ValueError(
+            f"Precision {enabled_precisions} not supported in the Dynamo Path"
+        )
+
+    if kwargs.get("ir", "dynamo") == "torch_compile":
+        custom_backend = create_backend(
+            precision=lower_precision,
+            debug=debug,
+            workspace_size=workspace_size,
+            min_block_size=min_block_size,
+            torch_executed_ops=torch_executed_ops,
+            pass_through_build_failures=pass_through_build_failures,
+            max_aux_streams=max_aux_streams,
+            version_compatible=version_compatible,
+            optimization_level=optimization_level,
+            use_python_runtime=use_python_runtime,
+            **kwargs,
+        )
+        model = torch.compile(gm, backend=custom_backend)
+        # Ensure compilation occurs by calling the function with provided inputs
+        model(*inputs)
+        return model
+
+    else:
+        settings = CompilationSettings(
+            debug=debug,
+            precision=lower_precision,
+            workspace_size=workspace_size,
+            min_block_size=min_block_size,
+            torch_executed_ops=torch_executed_ops,
+            pass_through_build_failures=pass_through_build_failures,
+            max_aux_streams=max_aux_streams,
+            version_compatible=version_compatible,
+            optimization_level=optimization_level,
+            use_python_runtime=use_python_runtime,
+        )
+
+        model = trace(gm, inputs, **kwargs)
+
+        if kwargs.get("use_capability_partitioner", None):
+            model = lower_model(model, inputs)
+            return _compile_module(model, inputs, settings)
+        else:
+            split_result = lower_model_using_trt_splitter(model, inputs)
+            trt_module = _compile_graph(split_result, inputs, settings)
+
+            return trt_module
+
+
+def create_backend(
```

Comment:
See above comment - potentially to delete this function.

Reply:
Done